packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -176,6 +177,14 @@ class Dismissible extends StatefulWidget {
   ///    be dismissed.
   final Map<DismissDirection, double> dismissThresholds;
 
+  /// Controls whether the widget can be dismissed by flinging in the indicated [direction].
+  ///
+  /// Flinging is treated as being equivalent to dragging almost to 1.0, so
+  /// flinging can dismiss an item past any threshold less than 1.0.
+  ///
+  /// This defaults to `true`
+  final bool allowFlinging;
```

Comment:
The main thing here that concerns me is a potential "technical debt". Several people would benefit from a boolean flag that determines whether flinging is allowed, but there's a lot of other scenarios that people might have requests for in the future.
Maybe someone wants a fling to be successful only when it reaches a certain velocity, or maybe based on whether it reaches a certain drag extent (either a number of pixels or a percentage of the width), or maybe based on a combination of the two.

The [gestures/](https://github.com/flutter/flutter/tree/b9b0784ea4089ee394c0ba92658bd4d25e466b9d/packages/flutter/lib/src/gestures) folder has a variety of classes that store details about various gestures (a few of them are used in this widget!)

Perhaps instead of an `allowFlinging` flag, we could have something like

```dart
final bool Function(DismissGestureDetails details) shouldDismiss;
```

And then developers could choose to either provide the `dismissThresholds` map or the `shouldDismiss` callback.

Reply:
Thank you for the review. This is indeed a very good point.
I have some concerns about the potential for confusion between the new `shouldDismiss` callback and the existing `confirmDismiss` 🤔 

Also, by implementing a simple/naive check of the percentage of the width dragged in the `shouldDismiss` callback to decide if we should start a dismiss action, we will implicitly break the current behaviour of having the flinging gesture enabled by default.

Reply:
Thanks for bringing up the potential confusion with the `confirmDismiss` name.

Fortunately, I don't think it would need to break current behavior. Maybe something like:

```dart
class _DismissibleState extends State<Dismissible> {
  bool shouldDismiss(DismissGestureDetails details) {
    late final _FlingGestureKind fling = _describeFlingGesture(details.velocity);

    return switch (widget.shouldDismiss) {
      final callback? => callback(details),
      null => _dragExtent >= _dismissThreshold || fling == _FlingGestureKind.forward,
    };
  }

  // ...
}
```